iothub/service/src/Authentication/AuthenticationMethodFactory.cs

```diff
@@ -12,17 +12,6 @@ public sealed class AuthenticationMethodFactory
     {
         internal static IAuthenticationMethod GetAuthenticationMethod(IotHubConnectionStringBuilder iotHubConnectionStringBuilder)
         {
-            if (!string.IsNullOrWhiteSpace(iotHubConnectionStringBuilder.DeviceId))
```

Comment:
a service connection string shouldn't have a device Id, so this change looks good